elixir/apps/web/lib/web/components/form_components.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Web.FormComponents do
+  @moduledoc """
+  Provides components that can be shared across forms.
+  """
+  use Phoenix.Component
+  use Web, :verified_routes
+  import Web.CoreComponents
+
+  @doc """
+  Conditionally renders form fields corresponding to a given provisioning strategy type.
+
+  ## Examples
+
+    <.provisioning_strategy_form form={%{
+      provisioning_strategy: "jit",
+      jit_user_filter_type: "email_allowlist",
+      jit_user_filter_value: "jamil@foo.dev,andrew@foo.dev",
+      jit_extract_groups: true
+    }} />
+
+    <.provisioning_strategy_form form={@form} />
+  """
+  attr :form, :map, required: true, doc: "The form to which this component belongs."
+
+  def provisioning_strategy_form(assigns) do
+    ~H"""
```

Comment:
I think we should not keep components like this one here, shared components are for things used everywhere in the app (eg. we should move `input`, `simple_form`, and other related components here) while this one is specific for identity provider views. Need to think about the better place to keep reusable things like this one but not polluting global namespace.

Reply:
Good catch, yeah, I'll move this somewhere else.

Reply:
Updated to keep these in a local `components.ex` file relative to the other views that depend on them.